phpBB/phpbb/controller/helper.php

```diff
@@ -22,53 +34,101 @@
+	protected $db;
+
+	/** @var dispatcher */
+	protected $dispatcher;
+
+	/** @var language */
+	protected $language;
+
 	/**
 	* Template object
-	* @var \phpbb\template\template
+	* @var template
 	*/
 	protected $template;
 
 	/**
 	* User object
-	* @var \phpbb\user
+	* @var user
 	*/
 	protected $user;
 
 	/**
 	* config object
-	* @var \phpbb\config\config
+	* @var config
 	*/
 	protected $config;
 
-	/* @var \phpbb\symfony_request */
+	/* @var symfony_request */
 	protected $symfony_request;
 
-	/* @var \phpbb\request\request_interface */
+	/* @var request_interface */
 	protected $request;
 
 	/**
-	 * @var \phpbb\routing\helper
+	 * @var routing_helper
 	 */
 	protected $routing_helper;
 
+	/** @var string */
+	protected $admin_path;
+
+	/** @var string */
+	protected $php_ext;
+
+	/** @var bool $sql_explain */
+	protected $sql_explain;
+
 	/**
 	 * Constructor
 	 *
-	 * @param \phpbb\template\template $template Template object
-	 * @param \phpbb\user $user User object
-	 * @param \phpbb\config\config $config Config object
-	 * @param \phpbb\symfony_request $symfony_request Symfony Request object
-	 * @param \phpbb\request\request_interface $request phpBB request object
-	 * @param \phpbb\routing\helper $routing_helper Helper to generate the routes
+	 * @param auth $auth Auth object
+	 * @param cache_interface $cache
+	 * @param manager $cron_manager
+	 * @param driver_interface $db Dbal object
+	 * @param dispatcher $dispatcher
+	 * @param language $language
+	 * @param template $template Template object
+	 * @param user $user User object
+	 * @param config $config Config object
+	 * @param symfony_request $symfony_request Symfony Request object
+	 * @param request_interface $request phpBB request object
+	 * @param routing_helper $routing_helper Helper to generate the routes
+	 * @param string $admin_path Admin path
+	 * @param string $php_ext PHP extension
+	 * @param bool $sql_explain Flag whether to display sql explain
 	 */
-	public function __construct(\phpbb\template\template $template, \phpbb\user $user, \phpbb\config\config $config, \phpbb\symfony_request $symfony_request, \phpbb\request\request_interface $request, \phpbb\routing\helper $routing_helper)
+	public function __construct(auth $auth, cache_interface $cache, manager $cron_manager, driver_interface $db,
+								dispatcher $dispatcher, language $language, template $template, user $user, config $config,
+								symfony_request $symfony_request, request_interface $request, routing_helper $routing_helper,
+								$admin_path, $php_ext, $sql_explain = false)
 	{
+		$this->auth = $auth;
+		$this->cache = $cache;
+		$this->cron_manager = $cron_manager;
+		$this->db = $db;
+		$this->dispatcher = $dispatcher;
+		$this->language = $language;
 		$this->template = $template;
 		$this->user = $user;
 		$this->config = $config;
 		$this->symfony_request = $symfony_request;
 		$this->request = $request;
 		$this->routing_helper = $routing_helper;
+		$this->admin_path = $admin_path;
```

Comment:
Since you're injecting so many new services anyway and have to amend tests already, any chance we can reorder the existing services aswell. As in, aphabetically.